server/tests/redis_test.rs

```diff
@@ -46,105 +34,50 @@ async fn redis_stores_and_returns_data_correctly() {
-            ..EdgeToken::from_str(TOKEN).unwrap()
-        },
-    }];
-
-    redis
-        .set_refresh_tokens(tokens.iter().collect::<Vec<&TokenRefresh>>())
+    let redis_persister = RedisPersister::new(&url).unwrap();
+    let mut project_specific_token =
+        EdgeToken::from_str("someproject:development.abcdefghijklmnopqr").unwrap();
+    project_specific_token.token_type = Some(TokenType::Client);
+    let mut wildcard_token = EdgeToken::from_str("*:development.mysecretispersonal").unwrap();
+    wildcard_token.token_type = Some(TokenType::Client);
+    redis_persister
+        .save_tokens(vec![project_specific_token, wildcard_token])
         .await
         .unwrap();
-    let returned_tokens = redis.get_refresh_tokens().await.unwrap();
-    assert_eq!(returned_tokens[0].token, tokens[0].token);
+    let saved_tokens = redis_persister.load_tokens().await.unwrap();
+    assert_eq!(saved_tokens.len(), 2);
 }
 
 #[tokio::test]
-async fn redis_store_marks_update_correctly() {
+async fn redis_saves_and_restores_token_refreshes_correctly() {
     let docker = Cli::default();
     let (_client, url, _node) = setup_redis(&docker);
-
-    let redis: RedisProvider = RedisProvider::new(&url).unwrap();
-
-    let token = EdgeToken {
-        status: TokenValidationStatus::Validated,
-        environment: Some("some-env-1".to_string()),
-        projects: vec!["default".to_string()],
-        ..EdgeToken::from_str(TOKEN).unwrap()
-    };
-
-    let entity_tag = EntityTag::new_weak("some-etag".to_string());
-    let token_refresh = TokenRefresh {
-        etag: None,
-        last_refreshed: None,
-        last_check: None,
-        token: token.clone(),
-    };
-
-    let tokens = vec![token_refresh.clone()];
-
-    redis
-        .set_refresh_tokens(tokens.iter().collect::<Vec<&TokenRefresh>>())
-        .await
-        .unwrap();
-
-    redis
-        .update_last_refresh(&token, Some(entity_tag.clone()))
+    let redis_persister = RedisPersister::new(&url).unwrap();
+    let edge_token = EdgeToken::from_str("someproject:development.abcdefghijklmnopqr").unwrap();
+
+    let mut token_refresh = TokenRefresh::new(edge_token.clone());
+    let now = Utc::now();
+    token_refresh.last_check = Some(now);
+    token_refresh.last_refreshed = Some(now);
+    token_refresh.etag = Some(EntityTag::new_weak("abcdefghijl".into()));
+    redis_persister
+        .save_refresh_targets(vec![token_refresh])
         .await
         .unwrap();
-
-    let found_token = redis
-        .get_refresh_tokens()
-        .await
-        .unwrap()
-        .get(0)
-        .unwrap()
-        .clone();
-
-    assert_eq!(found_token.etag, Some(entity_tag));
-    assert!(found_token.last_check.is_some());
-    assert!(found_token.last_refreshed.is_some());
+    let saved_refreshes = redis_persister.load_refresh_targets().await.unwrap();
+    assert_eq!(saved_refreshes.len(), 1);
+    assert_eq!(saved_refreshes.get(0).unwrap().token, edge_token);
 }
```

Comment:
Does this file still belong here? Maybe it's nicer to have it at the same level as `redis.rs` instead?

Reply:
I'd still prefer them here, this is cargo's approach to how integration tests are marked, so having them here makes them very easy to ignore when doing local dev. I think that's worth it since the unit tests run in about 50ms and these take nearly 3 seconds to run